test/Microsoft.Health.Fhir.R4.Tests.E2E/Rest/Audit/AuditTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Tests.Common;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.Tests.E2E.Rest.Audit
+{
+    /// <summary>
+    /// Provides R4 specific tests.
+    /// </summary>
+    [Trait(Traits.Category, Categories.Batch)]
+    public partial class AuditTests
+    {
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenABatch_WhenPost_ThenAuditLogEntriesShouldBeCreated()
+        {
+            if (!_fixture.IsUsingInProcTestServer)
+            {
+                // This test only works with the in-proc server with customized middleware pipeline
+                return;
+            }
+
+            List<(string expectedActions, string expectedPathSegments, HttpStatusCode expectedStatusCodes)> expectedList = new List<(string, string, HttpStatusCode)>
+            {
+                ("batch", string.Empty, HttpStatusCode.OK),
+                ("delete", "Patient/234", HttpStatusCode.OK),
```

Comment:
Is this allowed? I thought the same resource within a batch/transaction cannot have multiple actions (even though it's no-op)?

Reply:
True. Two entries for the same resource is in the list are to verify the audit entries(executing and executed) per resource. To avoid confusion, refactored the expectedList to contain HttpStatusCodes as null for executing entries